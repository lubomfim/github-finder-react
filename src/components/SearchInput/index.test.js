import React from 'react';
import { render, screen } from '@testing-library/react';

import { SearchInput } from './index';

const props = {
  onChange: () => {},
  disabled: true,
  value: 'userlogin',
};

const component = <SearchInput onChange={props.onChange} disabled={props.disabled} value={props.value} />;

it('should render correctly', () => {
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});

it('should be disabled', () => {
  render(component);
  expect(screen.getByRole('textbox')).toBeDisabled();
});
