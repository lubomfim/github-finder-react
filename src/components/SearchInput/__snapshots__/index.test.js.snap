// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  .c0 {
  background: #FFF;
  border: 1px solid #A7A7A7;
  font-size: 1.6rem;
  color: #000;
  padding: 20px 18px;
  outline: none;
  width: 100%;
  font-family: 'Segoe UI',sans-serif;
  box-shadow: 3px 3px 0px #000;
  -webkit-box-shadow: 3px 3px 0px #000;
  -moz-box-shadow: 3px 3px 0px #000;
}

.c0::-webkit-input-placeholder {
  color: #A7A7A7;
  font-size: 1.4rem;
}

.c0::-moz-placeholder {
  color: #A7A7A7;
  font-size: 1.4rem;
}

.c0:-ms-input-placeholder {
  color: #A7A7A7;
  font-size: 1.4rem;
}

.c0::placeholder {
  color: #A7A7A7;
  font-size: 1.4rem;
}

.c0:disabled {
  background-color: #eee;
  cursor: normal;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<input
    class="c0"
    disabled=""
    type="text"
    value="userlogin"
  />
</DocumentFragment>
`;
